From a9a9ebe9ee258d3c41035fef49210bd69249a073 Mon Sep 17 00:00:00 2001
From: "le.yang" <le.yang@amlogic.com>
Date: Tue, 25 Mar 2014 16:38:40 +0800
Subject: [PATCH 3881/5965] PD #87867: Resolve the out of av sync problem in
 html5  playing mode after seek operation, which is caused by ionvideo.c.

---
 drivers/amlogic/ionvideo/ionvideo.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/ionvideo/ionvideo.c b/drivers/amlogic/ionvideo/ionvideo.c
index 9956113ac51d..f046e5b8c1f3 100755
--- a/drivers/amlogic/ionvideo/ionvideo.c
+++ b/drivers/amlogic/ionvideo/ionvideo.c
@@ -608,7 +608,7 @@ static int vidioc_synchronization_dqbuf(struct file *file, void *priv, struct v4
 
     buf = container_of(vb, struct ionvideo_buffer, vb);
     if (dev->receiver_register) {
-        tsync_avevent_locked(VIDEO_START, buf->pts ? buf->pts : timestamp_pcrscr_get());
+        tsync_avevent_locked(VIDEO_START, buf->pts ? buf->pts : timestamp_vpts_get());
         dev->receiver_register = 0;
         d = 0;
     } else if (buf->pts) {
@@ -776,6 +776,7 @@ static int video_receiver_event_fun(int type, void* data, void* private_data) {
 
     if (type == VFRAME_EVENT_PROVIDER_UNREG) {
         dev->receiver_register = 0;
+        tsync_avevent(VIDEO_STOP, 0);
         printk("unreg:ionvideo\n");
     }else if (type == VFRAME_EVENT_PROVIDER_REG) {
         dev->receiver_register = 1;
-- 
2.19.0

